templates/fine-tune-function-calling/README.md

```diff
@@ -0,0 +1,419 @@
+# Fine-tuning for Function calling on custom data.
+
+**⏱️ Time to complete**: 5 hours for 7/8B models (11 hours for 13B, 27 hours for 70B)
+
+Function calling is an important capability of large language models. Connecting your model to external tools is at the heart of many LLM applications. In Anyscale Endpoints, you can use the [function calling API](https://docs.anyscale.com/preview/endpoints/text-generation/function-calling) to enable any model to use external tools. This is made possible [through JSON mode](https://www.anyscale.com/blog/anyscale-endpoints-json-mode-and-function-calling-features). However, it is beneficial to have *native* function calling capabilities in your model through fine-tuning on a relevant function calling dataset. JSON-mode-based function calling can only guarantee that the output is in the right schema, and can also be more expensive than a regular chat completion. However, fine-tuning on a function calling dataset can improve the model's capabilities with intent recognition (understanding when to call and when not to call a tool) and function call accuracy (employing the right function with accurate parameters) in addition to structured data formatting (formatting the function call json in the correct schema).  Fine-tuning would also be the only systematic way to improve performance on use-case-specific data. 
+
+In this example, we demonstrate fine-tuning on [Glaive's function calling dataset](https://huggingface.co/datasets/glaiveai/glaive-function-calling-v2?row=0) using Anyscale Endpoints. The dataset consists of about 113,000 examples of synthetically generated function calling data.  The dataset composition is given below:
+
+<p align="center">
+  <img src="./assets/distr_glaive_pie.png" alt="Distribution">
+</p>
+
+
+# Table of Contents
+1. [Data Preprocessing](#step-1-data-preprocessing)
+2. [Finetuning](#step-2-fine-tuning)
+3. [Serving](#step-3-serving)
+4. [Evaluation](#step-4-evaluation)
```

Comment:
In terms of the flow, this doesn't make much sense for me. Since we want the user to evaluate the model they fine-tuned in Step 2, serving should come before evaluation (otherwise what endpoint will `ANYSCALE_API_BASE` be)